docs/spec.md

```diff
@@ -712,18 +713,18 @@ defined as follows:
 * `flattened_ids(replica_groups)`
   if `channel_id > 0 and use_global_device_ids = true`.
 
-Afterwards, within each `process_group`:
+Afterwards, within each process_group and for each i in [0, size(operands)]:
```

Comment:
As I am looking at all the three all* operations together it makes sense to update them uniformly. 

remove `for each...`

```
* `gathered_operands...@receiver = [operands...@sender for sender in process_group]` for all
  `receiver` in `process_group`.
* `results...@process = concatenate(gathered_operands...@process, all_gather_dim)` for all
  `process` in `process_group`.
```